### PR TITLE
Changing base layer order

### PR DESCRIPTION
**Pull request by jterry75**
Partially Resolves: #78

### CODE REVIEW

**Comment by jterry75** [comment]
I have tested this e2e and verified that something such as:

`docker run --rm ubuntu cat /etc/resolv.conf` does return the correct name servers

File: service/gcs/core/gcs/storage.go [hunk 1]
```diff
@@ -311,23 +311,23 @@ func (c *gcsCore) mountLayers(id string, scratchMount *mountSpec, layers []*moun
 	// Mount the layer devices.
 	layerPaths := make([]string, len(layers)+1)
 	for i, layer := range layers {
-		layerPath := fmt.Sprintf("%s%d", layerPrefix, i)
+		layerPath := fmt.Sprintf("%s%d", layerPrefix, i+1)
```
Comment (soccerGB) [inline service/gcs/core/gcs/storage.go]: Interesting change.
Was the old code simply incorrect?

**Comment by jterry75** [comment]
Oops. Shifting the layer names broke the unit tests.

We actually could leave that part alone. I did that to maintain consistency with the array order in  `layerPaths` given that we shifted them all by `1`. If that is confusing I can put it back however that will  mean *layer0* is actually *layer1* in the overlay. Do you care one way over another?

Or I can just update the unit tests to read from the correct locations.

**Comment by gupta-ak** [comment]
I think it's fine if `layerPath := fmt.Sprintf("%s%d", layerPrefix, i+1)` gets changed to a `i` instead. The `layer[0]` is bascially as fake layer for our networking stuff, so it makes sense that `layer0` = first layer of real image overlay = `layer[1]`.

**Comment by gupta-ak** [comment]
It's because the baselayer was originally the bottom most read-only layer containing the `resolv.conf` for networking. If any image, like `ubuntu`, had the `/etc/resolv.conf` file in one of the layers, it would override our filled `resolv.conf` with whatever the layer had, breaking networking. Since now it's at the top, the correct `resolv.conf` won't get overwritten. See #78 for further discussion.

**Comment by soccerGB** [comment]
Ah, I didn't connect those two topics.  Thanks for the explanation

**Comment by jterry75** [comment]
Re-verified `docker run --rm ubuntu cat /etc/resolv.conf` with `layerPath := fmt.Sprintf("%s%d", layerPrefix, i)` and this passes unit tests again. Will work on another fix for `/sbin/udhcpc_config.script` separately

**Pull request merged**